consensus/src/leader.rs

```diff
@@ -76,22 +76,23 @@ where
         let mut reached_decided = false;
 
         let parent_leaf = if let Some(parent_view) = consensus.state_map.get(parent_view_number) {
-            match &parent_view.view_inner {
-                ViewInner::Leaf { leaf } => {
-                    if let Some(leaf) = consensus.saved_leaves.get(leaf) {
-                        if leaf.view_number == consensus.last_decided_view {
-                            reached_decided = true;
-                        }
-                        leaf
-                    } else {
-                        warn!("Failed to find high QC parent.");
-                        return self.high_qc;
+            if let Some(leaf) = parent_view.get_leaf_commitment() {
```

Comment:
nit: We could also use the let-else pattern here. I've pointed out other places where this could be done. Not sure what's more idiomatic. `let-else` forces control flow, which may be what we want.

```suggestion
            let Some(leaf) = parent_view.get_leaf_commitment() {
```